src/ray/raylet/node_manager.cc

```diff
@@ -1108,8 +1110,56 @@ void NodeManager::FinishAssignedTask(Worker &worker) {
   worker.AssignTaskId(TaskID::nil());
 }
 
-void NodeManager::ResubmitTask(const TaskID &task_id) {
-  RAY_LOG(WARNING) << "Task re-execution is not currently implemented";
+void NodeManager::HandleTaskReconstruction(const TaskID &task_id) {
+  // Retrieve the task spec in order to re-execute the task.
+  RAY_CHECK_OK(gcs_client_->raylet_task_table().Lookup(
+      JobID::nil(), task_id,
+      /*success_callback=*/
+      [this](ray::gcs::AsyncGcsClient *client, const TaskID &task_id,
+             const ray::protocol::TaskT &task_data) {
+        // The task was in the GCS task table. Use the stored task spec to
+        // re-execute the task.
+        const Task task(task_data);
+        ResubmitTask(task);
+      },
+      /*failure_callback=*/
+      [this](ray::gcs::AsyncGcsClient *client, const TaskID &task_id) {
+        // The task was not in the GCS task table. It must therefore be in the
+        // lineage cache. Use a copy of the cached task spec to re-execute the
+        // task.
+        const Task task = lineage_cache_.GetTask(task_id);
+        ResubmitTask(task);
+      }));
+}
+
+void NodeManager::ResubmitTask(const Task &task) {
+  // Actor reconstruction is turned off by default right now. If this is an
```

Comment:
just curious: is this a stage approach - aka. actor reconstruction will be enabled later?

Reply:
Yes, we're still thinking about how exactly actor reconstruction should be exposed to the user. I believe the current plan is to have actor reconstruction enabled by a flag from the frontend.